File: csharp/unittests/policy.tests/SimpleTextV1LexiconPolicyParser_buildX509FieldTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using FluentAssertions;
using Health.Direct.Common.Policies;
using Health.Direct.Policy.Impl;
using Health.Direct.Policy.X509;
using Xunit;

namespace Health.Direct.Policy.Tests
{
    public class SimpleTextV1LexiconPolicyParser_buildX509FieldTest
    {
        [Fact]
        public void TestBuildX509Field_SignatureAlgorithm_AssertBuilt()
        {
            var parser = new SimpleTextV1LexiconPolicyParser();

            IPolicyExpression field = parser.BuildX509Field("X509.Algorithm");
            Assert.NotNull(field);
            field.GetType().Should().Be(typeof(SignatureAlgorithmField));
        }

        [Fact]
        public void TestBuildX509Field_Signature_AssertFieldNotImplemented()
        {
            var parser = new SimpleTextV1LexiconPolicyParser();
            Assert.Throws<PolicyParseException>(() => parser.BuildX509Field("X509.Signature"));
        }

        [Fact]
        public void TestBuildX509Field_UnknownField_AssertNull()
        {
            var parser = new SimpleTextV1LexiconPolicyParser();
            IPolicyExpression field = parser.BuildX509Field("X509.Bogus");
            Assert.Null(field);
        }

    }
}

File: java/tags/direct-policy-1.0/src/test/java/org/nhindirect/policy/impl/SimpleTextV1LexiconPolicyParser_buildX509FieldTest.java
package org.nhindirect.policy.impl;

import org.nhindirect.policy.PolicyParseException;
import org.nhindirect.policy.x509.SignatureAlgorithmField;
import org.nhindirect.policy.x509.X509Field;

import junit.framework.TestCase;

public class SimpleTextV1LexiconPolicyParser_buildX509FieldTest extends TestCase
{
	public void testBuildX509Field_signatureAlgorithm_assertBuilt() throws Exception
	{
		final SimpleTextV1LexiconPolicyParser parser = new SimpleTextV1LexiconPolicyParser();
		
		final X509Field<?> field = X509Field.class.cast(parser.buildX509Field("X509.Algorithm"));
		assertNotNull(field);
		assertTrue(field instanceof SignatureAlgorithmField);
	}
	
	public void testBuildX509Field_signature_assertFieldNotImplemented() throws Exception
	{
		final SimpleTextV1LexiconPolicyParser parser = new SimpleTextV1LexiconPolicyParser();

		boolean exceptionOccured = false;
		
		try
		{
			X509Field.class.cast(parser.buildX509Field("X509.Signature"));
		}
		catch (PolicyParseException e)
		{
			exceptionOccured = true;
		}
		
		assertTrue(exceptionOccured);
	}
	
	public void testBuildX509Field_unknownField_assertNull() throws Exception
	{
		final SimpleTextV1LexiconPolicyParser parser = new SimpleTextV1LexiconPolicyParser();
		
		final X509Field<?> field = X509Field.class.cast(parser.buildX509Field("X509.Bogus"));
		assertNull(field);
	}
}
